gflownet/envs/ctorus.py

```diff
@@ -260,20 +402,74 @@ def step(
         """
         if self.done:
             return self.state, action, False
-        # If only possible action is eos, then force eos
-        # If the number of actions is equal to maximum trajectory length
+        # If only possible action is EOS (number of actions is equal to maximum
+        # trajectory length), then force EOS.
         elif self.n_actions == self.length_traj:
             self.done = True
             self.n_actions += 1
             return self.state, self.eos, True
-        # If action is eos, then it is invalid
+        # If anyway action is EOS, then it is invalid
         elif action == self.eos:
+            # [DEBUG] this point should never be reached. Consider removing this elif.
+            print(
+                "An EOS action has been sampled forward even though it is not allowed"
+            )
+            breakpoint()
```

Comment:
I would definitely remove the breakpoint, this might cause the code to get stuck (and in particular on the cluster, it might go unnoticed), if anything I'd raise an exception.

Reply:
It was removed [here](https://github.com/alexhernandezgarcia/gflownet/pull/193/commits/ffdc043cfad3bbf7ea7ace12e167356a7256bcde). Sorry, I had forgotten to push those changes.